packages/framework/client-logger/app-insights-logger/src/test/fluidAppInsightsLogger.spec.ts

```diff
@@ -25,13 +27,189 @@ describe("FluidAppInsightsLogger", () => {
+
+	beforeEach(() => {
+		appInsightsClient = new ApplicationInsights({
+			config: {
+				connectionString:
+					// (this is an example string)
+					"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+			},
+		});
+		trackEventSpy = spy(appInsightsClient, "trackEvent");
+	});
+
+	it("exclusive filter mode sends all events when no filters are defined", () => {
+		const logger = new FluidAppInsightsLogger(appInsightsClient, {
+			filterConfig: {
+				mode: "exclusive",
+			},
+		});
+
+		const perfCategoryEvent = {
+			category: TelemetryEventCategory.Performance,
+			eventName: "perfCategoryEventName",
+		};
+
+		const eventCount = 10;
+		for (let i = 0; i < eventCount; i++) {
+			logger.send(perfCategoryEvent);
+		}
+		// Expect all events to be sent in exclusive mode
+		sinonAssert.callCount(trackEventSpy, eventCount);
+	});
+
+	it("inclusive filter mode sends no events when no filters are defined", () => {
+		const logger = new FluidAppInsightsLogger(appInsightsClient, {
+			filterConfig: {
+				mode: "inclusive",
+			},
+		});
+
+		const perfCategoryEvent = {
+			category: TelemetryEventCategory.Performance,
+			eventName: "perfCategoryEventName",
+		};
+		for (let i = 0; i < 10; i++) {
+			logger.send(perfCategoryEvent);
+		}
+
+		// Expect no events to be sent
+		sinonAssert.callCount(trackEventSpy, 0);
+	});
+});
+
+describe("Telemetry Filter - Category filtering", () => {
+	let appInsightsClient: ApplicationInsights;
+	let trackEventSpy: Sinon.SinonSpy;
+
+	beforeEach(() => {
+		appInsightsClient = new ApplicationInsights({
+			config: {
+				connectionString:
+					// (this is an example string)
+					"InstrumentationKey=abcdefgh-ijkl-mnop-qrst-uvwxyz6ffd9c;IngestionEndpoint=https://westus2-2.in.applicationinsights.azure.com/;LiveEndpoint=https://westus2.livediagnostics.monitor.azure.com/",
+			},
+		});
+		trackEventSpy = spy(appInsightsClient, "trackEvent");
+	});
+
+	it("exclusive filter mode sends ALL events except those that match category filters", () => {
+		const logger = new FluidAppInsightsLogger(appInsightsClient, {
+			filterConfig: {
+				mode: "exclusive",
+				filters: [
+					{
+						category: TelemetryEventCategory.Performance,
+					},
+					{
+						category: TelemetryEventCategory.Generic,
+					},
+				],
+			},
+		});
+
+		const errorCategoryEvent = {
+			category: TelemetryEventCategory.Error,
+			eventName: "errorCategoryEventName",
+		};
+		const perfCategoryEvent = {
+			category: TelemetryEventCategory.Performance,
+			eventName: "perfCategoryEventName",
+		};
+		const genericCategoryEvent = {
+			category: TelemetryEventCategory.Generic,
+			eventName: "genericCategoryEventName",
+		};
+
+		for (let i = 0; i < 10; i++) {
+			logger.send(perfCategoryEvent);
+			logger.send(errorCategoryEvent);
+			logger.send(genericCategoryEvent);
+		}
```

Comment:
I think we can skip the loop here. It made sense for sampling tests to generate enough volume that sampling would come into play; here I think a single send() of each kind is enough.

Reply:
You're right. Updated these tests.